pkg/api/mlflow/dao/repositories/param.go

```diff
@@ -72,32 +85,41 @@ func (r ParamRepository) CreateBatch(ctx context.Context, batchSize int, params
 
 // findConflictingParams checks if there are conflicting values for the input params. If a key does not
 // yet exist in the db, or if the same key and value already exist for the run, it is not a conflict.
-// If the key already exists for the run but with a different value, it is a conflict. Conflicting keys
-// are returned.
-func (r ParamRepository) findConflictingParams(
-	ctx context.Context,
-	params []models.Param,
-) ([]map[string]string, error) {
-	var paramsInError []map[string]string
-	if err := r.db.WithContext(ctx).
-		Raw(fmt.Sprintf(`
-		    WITH new(key, value, run_uuid) AS (VALUES %s)
-		    SELECT current.key as key, current.value as old_value, new.value as new_value
-		    FROM params AS current
-		    INNER JOIN new ON new.run_uuid = current.run_uuid AND new.key = current.key
-		    WHERE new.value != current.value
-                `, prepareSqlValues(params))).
-		Scan(&paramsInError).Error; err != nil {
-		return nil, eris.New("error fetching params from db")
+// If the key already exists for the run but with a different value, it is a conflict. Conflicts are returned.
+func findConflictingParams(tx *gorm.DB, params []models.Param) ([]paramConflict, error) {
+	var conflicts []paramConflict
+	if err := tx.Raw(
+		`WITH new(key, value, run_uuid) AS (VALUES `+makeSqlPlaceholders(params)+`)
```

Comment:
could you please use `fmt.Sprintf` here to improve readability?

Reply:
i started with that but it seemed like this was clearer -- since we use a values array for sql parameter replacement, it seemed weird to have first one value replacement for the template string (sql), then another

Reply:
looks better w/ separate statement to make the sql template, please take a look

Reply:
yes, it looks good now!